OctopusDeploy/step-package-template
# Updates to latest api and cli and publishes via Octopus

This PR updates the template to be consistent with the latest setup for building and publishing in order so it can serve as a template for new step packages.

Specifically:
- Update to the latest `step-api` and `step-package-cli`
- Publish via Octopus
- Build every night and publish to the nightly feed
- Include linting as part of build
- Ensure that automated dependency management using renovate uses the correct `pnpm` version
